connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java

```diff
@@ -128,13 +125,10 @@ private void doClose() {
     private void doRun() {
         try {
             synchronized (this) {
-                if (stopping.get())
+                if (stopping)
                     return;
 
-                if (targetState.get() == TargetState.PAUSED)
-                    statusListener.onPause(id);
-                else
-                    statusListener.onStartup(id);
+                statusListener.onStartup(id);
```

Comment:
And importantly, I think this change will cause the status to change briefly after a rebalance -- giving the `onStartup` call will set the status to `RUNNING` and then the subsequent `onPause` will set it back to `PAUSED`.